metrics/web/docs/features/public_dashboard/01_public_dashboard.md

```diff
@@ -137,3 +158,187 @@ Once we've chosen the implementation approach, let's consider the components dia
+
+![Public Dashboard Architecture Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_architecture_class.puml)
+
+As we can see on the diagram, we need to implement the following classes: 
+- [PublicDashboardFeatureConfigViewModel](#PublicDashboardFeatureConfigViewModel)
+- [UserProfileViewModel](#UserProfileViewModel)
+- [SignInAnonymouslyUseCase](#SignInAnonymouslyUseCase)
+- [AuthState](#AuthState)
+
+Let's consider each class in a bit more detail: 
+
+#### PublicDashboardFeatureConfigViewModel
+> Explain the purpose and responsibility of the class.
+
+Since we should sign in the user anonymously after the app initializing, we should be able to access the feature state from the UI. As the [Presentation Layer Architecture](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/02_presentation_layer_architecture.md#view-model) document states, we should create a view model to be able to access the data from the `state` on the UI. So, for this purpose, we should create a view model representing a public dashboard feature state to be able to work with this state value, for example, on the `loading page`.
+
+#### UserProfileViewModel
+> Explain the purpose and responsibility of the class.
+
+Since we want to disable some features on the UI if the user is anonymous, we should be able to access the user profile from the UI. As the [Presentation Layer Architecture](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/02_presentation_layer_architecture.md#view-model) document states, we should create a view model to be able to access the data from the `state` on the UI. So, we should create a view model representing a user profile to disable the `Project Groups` page and provide the ability to `Sign In` to the application for the anonymous users.
+
+#### SignInAnonymouslyUseCase
+> Explain the purpose and responsibility of the class.
+
+Since the public dashboard feature implies logging in as an anonymous user, we should create a use case needed to log in to the application using the anonymous user. 
+
+#### AuthState
+> Explain the purpose and responsibility of the class.
+
+Since we should know whether the user signed in anonymously or using sign in providers or credentials, or logged out, we should provide an enum, which should contain all the described states. 
+
+### User Interface
+> How users will interact with the feature (API, CLI, Graphical interface, etc.).
+
+Since the non-authorized users should not be able to access the `Project Groups` page, it should be hidden from the user menu for such users. Also, there should be an ability to sign in for anonymous users, so we should change the `Logout` button from the user menu to `Sign in` one. 
```

Comment:
:readable:
```suggestion
As we've mentioned above, the non-authorized users should not be able to access the `Project Groups` page. Because of this, we should hide this option from the user menu for anonymous users. Also, there should be an ability to sign in for anonymous users, so we should change the `Logout` button from the user menu to the `Sign in` one. 
```